src/plugins/data_source_management/public/components/data_source_table/data_source_table.tsx

```diff
@@ -255,10 +255,12 @@ export const DataSourceTable = ({ history }: RouteComponentProps) => {
 
   const setDefaultDataSource = async () => {
     try {
-      for (const dataSource of selectedDataSources) {
-        if (getDefaultDataSourceId(uiSettings) === dataSource.id) {
-          await setFirstDataSourceAsDefault(savedObjects.client, uiSettings, true);
-        }
+      const restDataSourceIds = dataSources
+        .filter((dataSource) => !selectedDataSources.includes(dataSource))
+        .map((item) => item.id);
+      if (restDataSourceIds.length === 0) uiSettings.remove('defaultDataSource');
+      if (!restDataSourceIds.includes(uiSettings.get('defaultDataSource'))) {
+        uiSettings.set('defaultDataSource', restDataSourceIds[0]);
```

Comment:
Hi Emma. in the onClickDelete(), we have setSelectedDataSources([]); I understand the state changes are asynchronous and react tends to batch multiple state updates. But there is still an possibility the selectedDataSource has been updated to []. Also, if we remove the setFirstDataSourceAsDefault, we can remove the async. 
```
    deleteMultipleDataSources(savedObjects.client, selectedDataSources)
      .then(() => {
        setSelectedDataSources([]);
        // Fetch data sources
        fetchDataSources();
        setConfirmDeleteVisible(false);
        // Check if default data source is deleted or not.
        // if yes, then set the first existing datasource as default datasource.
        setDefaultDataSource();
      })
```

Reply:
Let's sync this offline maybe?